flake.nix

```diff
@@ -22,6 +22,130 @@
       rust-overlay,
     }:
     let
+      niri-package =
+        {
+          lib,
```

Comment:
it's easy to miss in the diff but when moving this package declaration up here, i've also separately added the `lib` parameter so that comes from `pkgs.callPackage` instead of our outermost `inherit (nixpkgs) lib;`.